Fixed bug where multiple configs introduces undefined in the merged configs for a certain module

nkimaina opened the pull request.

brandones commented on package.json:
```diff
@@ -6,6 +6,7 @@
   "scripts": {
     "clean": "rimraf dist",
     "start": "webpack-dev-server",
+    "start:https": "webpack-dev-server --https",
```
I think the convention in our other modules is to call this `start-https`

brandones commented:
Interesting! Could you write a test for this? I'm not totally sure what the problem is.

nkimaina commented:
I have updated the unit test to reflect the test cases that produces this bug. I didn't the description clear sorry. It occurs when you have multiple configs defined. 
The unit test in question is:  
```npx jest -t 'works for multiple modules'```
The following screen shots will show you the bugs:
![Screenshot 2020-02-24 at 11 07 47](https://user-images.githubusercontent.com/11956424/75138403-2e318a00-56fb-11ea-868c-577c4d50676a.png)

<img width="499" alt="Screenshot 2020-02-24 at 10 40 17" src="https://user-images.githubusercontent.com/11956424/75138512-7781d980-56fb-11ea-8083-73a80d46a261.png">

<img width="627" alt="Screenshot 2020-02-24 at 11 40 52" src="https://user-images.githubusercontent.com/11956424/75138534-823c6e80-56fb-11ea-8135-0b4dfe1fa746.png">

That said. I have added the test scenario that produces it. 
See the example in our ampath repo:  
https://github.com/AMPATH/ampath-esm-patient-chart-widgets/pull/3/commits/eb23cad12be83096259b4997884ff74c6bb2a471

brandones commented on src/module-config/module-config.test.ts:
```diff
@@ -118,6 +119,101 @@ describe("getConfig", () => {
+    // TEST CASE 2: One that bugs out for multiple modules: see conversation on:
+    // https://github.com/openmrs/openmrs-esm-module-config/pull/18
+    Config.clearAll();
+    Config.defineConfigSchema("@openmrs/esm-patient-chart", {
```
This is a very complex example of the problem. It doesn't make it clear what exactly the issue is. Could you please reduce it to a [minimal working example](https://stackoverflow.com/help/minimal-reproducible-example)?

nkimaina commented:
The problem is the following line ends up with [ object, undefined]. It's easier to fix this here by eliminating the undefined (and end up with [object]) than try to fix the consequences downstream. 
I am not able to login into our distro because the bug breaks the app. 
https://github.com/openmrs/openmrs-esm-module-config/blob/master/src/module-config/module-config.ts#L108

brandones commented:
Thanks for finding and fixing this!
